transaction.go

```diff
@@ -74,17 +74,24 @@ type Transaction struct {
 
 	// PayloadSignatures is a list of signatures generated by the proposer and authorizer roles.
 	//
-	// A payload signature is generated over the inner portion of the transaction (payload).
+	// A payload signature is generated over the inner portion of the transaction (TransactionDomainTag + npayload).
 	//
 	// You can find more information about transaction signatures here: https://docs.onflow.org/concepts/transaction-signing/#anatomy-of-a-transaction
 	PayloadSignatures []TransactionSignature
 
 	// EnvelopeSignatures is a list of signatures generated by the payer role.
 	//
-	// A payload signature is generated over the outer portion of the transaction (payload + payloadSignatures).
+	// A envelope signature is generated over the outer portion of the transaction (TransactionDomainTag + payload + payloadSignatures).
 	//
 	// You can find more information about transaction signatures here: https://docs.onflow.org/concepts/transaction-signing/#anatomy-of-a-transaction
 	EnvelopeSignatures []TransactionSignature
+
+	// TransactionDomainTag is prepended to the transaction message before it is signed.
+	//
+	// The transaction domain tag is not encoded together with the transaction. The default transaction domain tag is `paddedDomainTag("FLOW-V0.0-transaction")`
+	//
+	// You can find more information about transaction domain tag here: https://docs.onflow.org/concepts/transaction-signing/#anatomy-of-a-transaction
+	TransactionDomainTag []byte
```

Comment:
`TransactionDomainTag` is a protocol-wide constant, rather than a transaction-specific field. I don't think it should be added to the transaction structure. 

The constant can be used directly in `SignEnvelope` and `SignPayload`.

Reply:
the `TransactionDomainTag` has an associated version: `paddedDomainTag("FLOW-V0.0-transaction")`. Is there no chance that in the future if this version changes, someone will want to use the new version before the sdk is updated (or without updating the sdk)?

Reply:
It's still not clear to me what the versioning will be used for (versions of the transaction model, versions of Flow forks, ..). For now the FMV only supports a unique static domain tag when verifying signatures, and not a dynamic one read on the transaction. If we were to enable multiple version support, we could have a specific field for for the version in the transaction structure instead of the entire tag, and that field will be checked by the FVM (and collection nodes) for transaction validity.  

Btw, I think it's a good idea to make the SDK re-use the [tag used by the FVM ](https://github.com/onflow/flow-go/blob/master/model/flow/constants.go#L43).

Reply:
> It's still not clear to me what the versioning will be used for (versions of the transaction model, versions of Flow forks, ..). For now the FMV only supports a unique static domain tag when verifying signatures, and not a dynamic one read on the transaction. If we were to enable multiple version support, we could have a specific field for for the version in the transaction structure instead of the entire tag, and that field will be checked by the FVM (and collection nodes) for transaction validity.

I understand. I will change it as you propose.

> Btw, I think it's a good idea to make the SDK re-use the tag used by the FVM .

`flow-go-sdk` does not depend on flow-go. In fact flow-go depends on the `flow-go-sdk`. So If anything, I could change `flow-go` to use the tag defined in the sdk, but that does not seem like a good idea.

Reply:
Hm right, the only dependency I know in the opposite way  is `flow-go-sdk/crypto` depending on `flow-go/crypto`. 
I'll add the change on `flow-go` then.